### PR TITLE
client requested zipped block but didn't decompress it so now requesting unzipped block

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
The client used to request zlib compressed data, but no code existed to decompress it. Removed the flag for zlib compression support. Entire conversation with the hub now happens through plain text.

Also improved the "$MyINFO" regex.
There are some wild threads that can still be fixed.

Using your code in our project [recobot](https://github.com/h4ck3rk3y/recobot). Thank you!

/cc @kaustubh-karkare

### CODE REVIEW

File: pydc_client.py [hunk 1]
```diff
@@ -381,7 +381,7 @@ def server_handler(self,data,info,args): # Interacts with the DC, responding to
 						if nick=="": continue
 						self._nicklist[nick]["bot"] = (True if nick in bots else False)
 				elif x[0]=="$MyINFO":
-					nick,desc,conn,flag,email,share = re.findall("^\$MyINFO \$ALL ([^ ]*) ([^\$]*)\$ \$([^\$]*)([^\$])\$([^\$]*)\$([^\$]*)\$$",data)[0]
+					nick,desc,conn,flag,email,share = re.findall("^\$MyINFO \$ALL ([^ ]*) ([^\$]*)\$.\$([^\$]*)([^\$])\$([^\$]*)\$([^\$]*)\$",data)[0]
```
Comment (kaustubh-karkare) [inline pydc_client.py:384]: Could you explain why this change was needed?
It's been a while since I've looked into this code, so I don't remember the exact format of this message.

File: pydc_client.py [hunk 2]
```diff
@@ -337,7 +337,7 @@ def server_handler(self,data,info,args): # Interacts with the DC, responding to
 			if data[0]=="<" and self._mainchat: self._mainchat(data+"\n")
 			elif data[0]=="$":
 				x = data.split()
-				if x[0]=="$Lock": self._socket.send("$Supports UserCommand UserIP2 TTHSearch ZPipe0 GetZBlock |$Key "+self.lock2key(x[1])+"|$ValidateNick "+self._config["nick"]+"|")
+				if x[0]=="$Lock": self._socket.send("$Supports UserCommand UserIP2 TTHSearch GetZBlock |$Key "+self.lock2key(x[1])+"|$ValidateNick "+self._config["nick"]+"|")
```
Comment (kaustubh-karkare) [inline pydc_client.py:340]: I'm fine with this :)

File: pydc_client.py (hunk 1 above)
Comment (h4ck3rk3y) [inline pydc_client.py:384]: There are clients that that don't put a space where I put the `.`. Either there is no space or there is a random alphabet. While testing the bot was crashing for a particular string, which I am not being able to find now.

Comment (kaustubh-karkare) [inline pydc_client.py:384]: Well, a specific example would be useful, but I'll trust you and accept this.

Comment (h4ck3rk3y) [inline pydc_client.py:384]: Kept pressing the up arrow key in ipython and finally found the string.
`$MyINFO $ALL bourne99 <ApexDC++ V:1.6.0,M:P,H:1/0/0,S:3>$P$0.11$$125525478604$`
